Fix various warnings in tests for v3.11.3

maresb opened the pull request:
**Thank your for opening a PR!**

Depending on what your PR does, here are a few things you might want to address in the description:
+ [X] what are the (breaking) changes that this PR makes?
+ [X] important background, or details about the implementation
+ [X] are the changes—especially new features—covered by tests and docstrings?
+ [X] [linting/style checks have been run](https://github.com/pymc-devs/pymc3/wiki/PyMC3-Python-Code-Style)
+ [X] [consider adding/updating relevant example notebooks](https://github.com/pymc-devs/pymc-examples)
+ [X] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

* There are no breaking changes, I hope.
* Background is that I was annoyed by superfluous warnings which show up during the test suite and wanted to clean them up.
* There are no new features, so no additional tests/docstrings
* I ran pre-commit.
* No example notebooks updated.
* I'm not sure this deserves mention in the release notes...

maresb committed:
Fix deprecation warning for np.int
https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

maresb committed:
Fix return_inferencedata warnings in tests

maresb commented:
NOTE: I still need to run pytest to make sure I didn't break anything. I was hoping it would be automatically triggered in CI.

ricardoV94 commented:
> NOTE: I still need to run pytest to make sure I didn't break anything. I was hoping it would be automatically triggered in CI.

Github has a new policy where the CI has to be manually approved for first time contributors. It seems someone already did it.

ricardoV94 commented:
It seems like the changes broke the `TestScalarParameterSamples` in `test_distributions_random.py` for discrete distributions.

maresb commented:
Thanks @ricardoV94!

I also reproduced this locally, and was quite perplexed. Even moreso when I reverted to `pymc-devs:v3.11.3` (without my changes) and reproduced the errors there.

Could it be that these errors are resulting from a previous commit?

Sayam753 commented:
Should the target branch be `v3` instead of `v3.11.3`?

Because I think v3.11.3 has not released yet!

maresb commented:
@Sayam753 I don't know. But maybe this should be fixed before the release?

Please see #4808.

maresb commented:
Ah, one of my commits coincides exactly with #4741.

The remaining commits are however still relevant.

What's the status of the `v3.11.3` branch? Should I rebase on `v3` instead?

ricardoV94 commented:
> What's the status of the `v3.11.3` branch? Should I rebase on `v3` instead?

I think yes, that's our main V3 branch.

Edit: The PR itself should target it

maresb commented on pymc3/__init__.py:
```diff
@@ -13,7 +13,7 @@
 
 # pylint: disable=wildcard-import
-__version__ = "3.11.2"
+__version__ = "3.11.3"
```
Hmm, where did this come from? Some artifact from me having changed the base branch?

maresb replied:
Now it's fixed after a rebase.

ricardoV94 commented:
You might need to rebase on V3 branch and drop unwanted commits from 3.11.3 branch

maresb commented:
Ok, I rebased, dropped the stray commit which bumped the version number, dropped the redundant commit, and squashed my two `return_inferencedata` commits.

Now everything should be clean, and it looks clean. There are now a total of two commits on `v3`:
* Fix deprecation warning for `np.int`
* Fix `return_inference` warnings in tests

ricardoV94 commented:
Thanks @maresb, looks good. The tests will still fail due to #4810 issue you found.

Are you interested in fixing that either in here or on a separate PR?

maresb commented:
@ricardoV94 I have to work my real job this week, so the earliest I could touch #4810 would be this weekend, optimistically.

ricardoV94 commented:
> @ricardoV94 I have to work my real job this week, so the earliest I could touch #4810 would be this weekend, optimistically.

Sure. Let's see if anyone jumps on this one before, and if not your help next weekend is still much appreciated!
